### PR TITLE
Update OCP check script and upstream support with the prow changes

### PR DESCRIPTION
**Pull request by mpryc**
The PROW CI jobs were updated to use only latest OCP release for their runs. The documentation and the check openshift version script needs to be changed as well.

PROW PR: https://github.com/openshift/release/pull/44348

### CODE REVIEW

File: scripts/check_openshift_version.py
Comment (mateusoliveira43) [inline scripts/check_openshift_version.py:29]: would change to `OCP_VERSIONS_URL`

File: docs/UpstreamSupport.md
Comment (mateusoliveira43) [inline docs/UpstreamSupport.md:3]: ...stable minor **versions** of the... -> ...stable minor **version** of the...

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
Also need to remove old references in README.md and .github/ISSUE_TEMPLATE/bug.yml

I would also add a deprecated warning (just a comment is enough) to Makefile E2E test scenario 2 command https://github.com/dora-metrics/pelorus/blob/master/Makefile#L126

**Comment by mateusoliveira43** [comment]
/hold
wait the other PR merge

**Comment by mpryc** [comment]
> Also need to remove old references in README.md and .github/ISSUE_TEMPLATE/bug.yml
> 
> I would also add a deprecated warning (just a comment is enough) to Makefile E2E test scenario 2 command https://github.com/dora-metrics/pelorus/blob/master/Makefile#L126

I left the issue list of older OCP releases. It's a bug entry and we don't force user to use X or Y version of OCP + we should allow to log against already existing deployments.

I don't think we should deprecate scenario-2, we just don't run them in the periodics.

**Review by mateusoliveira43 — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
I think the regex for PROW is working, it did not trigger for your newest commit that does not touch the files delacred in it :smile: 

But I think we left this there https://github.com/openshift/release/blob/master/core-services/prow/02_config/dora-metrics/pelorus/_prowconfig.yaml#L19 we should remove it right? 

(if you want, you can bypass branch protection and merge this PR)

**Comment by mpryc** [comment]
> I think the regex for PROW is working, it did not trigger for your newest commit that does not touch the files delacred in it 😄
> 
> But I think we left this there https://github.com/openshift/release/blob/master/core-services/prow/02_config/dora-metrics/pelorus/_prowconfig.yaml#L19 we should remove it right?
> 
> (if you want, you can bypass branch protection and merge this PR)

I think it's because my PR was after the above was merged, so it got old required jobs.

**Comment by mpryc** [comment]
Merging as it previously passed checks.

**Pull request merged**
